webapp/packages/core-blocks/src/useListKeyboardNavigation.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { useEffect, useState } from 'react';
+
+import { EventContext } from '@cloudbeaver/core-events';
+
+export const EventKeyboardNavigationFlag = EventContext.create('useListKeyboardNavigation');
+
+export function useListKeyboardNavigation<T extends HTMLElement>(elementsSelector = '[tabindex]:not(:disabled)'): (obj: T | null) => void {
+  const [ref, setRef] = useState<T | null>(null);
+
+  useEffect(() => {
+    if (!ref) {
+      return;
+    }
+
+    const getFocusableElements = () => {
+      const allFocusableElements = Array.from(ref.querySelectorAll(elementsSelector)) as HTMLElement[];
+      return allFocusableElements;
+    };
+
+    // Function to reset tabindex on all elements and set it to 0 on aria-selected="true"
+    const resetTabindex = () => {
+      const focusableElements = getFocusableElements();
+      focusableElements.forEach(el => {
+        if (el.getAttribute('aria-selected') === 'true') {
+          el.setAttribute('tabindex', '0');
+        } else {
+          el.setAttribute('tabindex', '-1');
+        }
```

Comment:
I suppose this set of elements can include elements without `aria-selected` attribute. Is this okay to set `tabIndex` for them also to `-1`? Or it is supposed to be only for elements with `aria-selected` attribute?

Reply:
this is okay, if element don't have this attribute then tabindex=-1 is okay, but here more problems over -1 and 0 because sometimes you can't restore initial value (when you don't have aria-selected or when element was tabindex=0 initially)

but for now seems that it works, maybe we will found bugs related to this function later